internal/resolution/entitysources/catalogdsource.go

```diff
@@ -0,0 +1,142 @@
+package entitysources
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/operator-framework/deppy/pkg/deppy"
+	"github.com/operator-framework/deppy/pkg/deppy/input"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/operator-framework/operator-registry/alpha/property"
+
+	catalogd "github.com/operator-framework/catalogd/pkg/apis/core/v1beta1"
+)
+
+// catalogdEntitySource is a source for(/collection of) deppy defined input.Entity, built from content
+// made accessible on-cluster by https://github.com/operator-framework/catalogd.
+// It is an implementation of deppy defined input.EntitySource
+type catalogdEntitySource map[deppy.Identifier]input.Entity
+
+func NewCatalogdEntitySource(ctx context.Context, client client.Client) (catalogdEntitySource, error) {
+	entities := catalogdEntitySource{}
+	bundleMetadatas, packageMetdatas, err := onClusterMetadata(ctx, client)
+	if err != nil {
+		return nil, err
+	}
+	for _, bundle := range bundleMetadatas.Items {
+		props := map[string]string{}
+
+		for _, prop := range bundle.Spec.Properties {
+			switch prop.Type {
+			case property.TypePackage:
+				type pkg struct {
+					Name    string `json:"packageName"`
+					Version string `json:"version"`
+				}
+				tmp := pkg{}
+				if err := json.Unmarshal(prop.Value, &tmp); err != nil {
+					return nil, err
+				}
+				pkgValue, err := jsonMarshal(tmp)
+				if err != nil {
+					return nil, err
+				}
+				props[property.TypePackage] = string(pkgValue)
+			}
+		}
+
+		imgValue, err := jsonMarshal(bundle.Spec.Image)
+		if err != nil {
+			return nil, err
+		}
+		props["olm.bundle.path"] = string(imgValue)
```

Comment:
Should this refer to the package constant?

Reply:
Good question. This PR is highlighting to me that we need some specific focus on the translation of FBC to what deppy needs. It seems there's an (unnecessary?) impedance mismatch that is necessitating some of this.

Reply:
To expand on my previous comment, this is not an actual FBC property. This is something sort of only happening at the deppy layer. I _think_ this will go away when @perdasilva removes the entity source concept and replaces it with a variable source.